.github/cd.yaml

```diff
@@ -0,0 +1,107 @@
+name: Deploy to NPM and GitHub Package Registry
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    env:
+      GTP_TOKEN: ${{ secrets.CROSS_REPO_TOKEN }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          registry-url: 'https://registry.npmjs.org'
+      - run: npm ci
+      - name: Create empty .env file
+        run: touch .env
+      - run: npm test
+      - run: npm run build
+      - name: Publish to NPM
+        run: |
+          echo "//registry.npmjs.org/:_authToken=${NODE_AUTH_TOKEN}" > .npmrc
+
+          # Read version from package.json
+          VERSION=$(node -p "require('./package.json').version")
+          echo "Current package version: $VERSION"
```

Comment:
we have been using this action instead in the other repos, thoughts?

https://github.com/marketplace/actions/npm-publish

Reply:
I did not think of this. Yes, we can consider it.